### PR TITLE
Simplify `F.sum` test

### PR DESCRIPTION
**Pull request by aksub99**
Simplified F.sum test using testing.FunctionTestCase.
Ref. #6071

### CODE REVIEW

**Comment by aksub99** [comment]
@Crissman The outputs of forward() and forward_expected() seem to differ by a small amount, therefore causing the error. Could you please review the code and suggest why this might be so?

File: tests/chainer_tests/functions_tests/math_tests/test_sum.py [hunk 1]
```diff
@@ -17,58 +17,41 @@
...
+        'use_cudnn': ['never', 'always'],
+        'cuda_device': [0, 1],
+    })
+    # ChainerX tests
+    + [
+        {'use_chainerx': True, 'chainerx_device': 'native:0'},
+        {'use_chainerx': True, 'chainerx_device': 'cuda:0'},
+        {'use_chainerx': True, 'chainerx_device': 'cuda:1'},
+    ])
+class TestSum(testing.FunctionTestCase):
```
Comment (crcrpar) [inline tests/chainer_tests/functions_tests/math_tests/test_sum.py]: It seems that this class lacks the relaxation of tolerance when self.dtype == numpy.float16.
I think this absence of tolerance for float16 relates to failures.

Comment (aksub99) [reply tests/chainer_tests/functions_tests/math_tests/test_sum.py]: Thanks @crcrpar . That managed to fix it !

**Comment by aksub99** [comment]
The failures seem to be caused due to `test_average_pooling_nd` . 
@Crissman @crcrpar  Please review and confirm. Thanks in advance!

File: tests/chainer_tests/functions_tests/math_tests/test_sum.py [hunk 2]
```diff
@@ -2,73 +2,57 @@
...
     'dtype': [numpy.float16, numpy.float32, numpy.float64],
 }))
-class TestSum(unittest.TestCase):
+@testing.fix_random()
+@testing.inject_backend_tests(
+    None,
+    # CPU tests
+    [
+        {},
+        {'use_ideep': 'always'},
```
Comment (crcrpar) [inline tests/chainer_tests/functions_tests/math_tests/test_sum.py]: It doesn't seem that [`F.sum`](https://github.com/chainer/chainer/blob/master/chainer/functions/math/sum.py#L11) use ideep so I don't think this case is necessary.

File: tests/chainer_tests/functions_tests/math_tests/test_sum.py [hunk 3]
```diff
@@ -2,73 +2,57 @@
...
+    ]
+    # GPU tests
+    + testing.product({
+        'use_cuda': [True],
+        'use_cudnn': ['never', 'always'],
```
Comment (crcrpar) [inline tests/chainer_tests/functions_tests/math_tests/test_sum.py]: Like the above case, It doesn't seem that cudnn is used in `F.sum`.

Comment (aksub99) [reply tests/chainer_tests/functions_tests/math_tests/test_sum.py]: Thanks for reviewing @crcrpar ! Have made the required changes.

**Comment by Crissman** [comment]
LGTM. Jenkins, test this please.

**Comment by Crissman** [comment]
Merging!

**Pull request merged**